### ISSUE TITLE
Add a FlexibleViewHolder

### ISSUE THREAD

**Issue by danielpassos** [davideas/FlexibleAdapter#14]
I was think about add a new utility class `FlexibleViewHolder` it will help developers it alwasy need to implements the `View.OnClickListener` and `View.OnLongClickListener`

```Java
public abstract class FlexibleViewHolder extends RecyclerView.ViewHolder
        implements View.OnClickListener, View.OnLongClickListener {

    private final FlexibleAdapter adapter;
    private final OnListItemClickListener onListItemClickListener;

    public FlexibleViewHolder(View itemView,FlexibleAdapter adapter,
                              OnListItemClickListener onListItemClickListener) {
        super(itemView);
        this.adapter = adapter;

        this.onListItemClickListener = onListItemClickListener;

        this.itemView.setOnClickListener(this);
        this.itemView.setOnLongClickListener(this);
    }

    @Override
    public void onClick(View view) {
        if (onListItemClickListener.onListItemClick(getAdapterPosition())) {
            toggleActivation();
        }
    }

    @Override
    public boolean onLongClick(View view) {
        onListItemClickListener.onListItemLongClick(getAdapterPosition());
        toggleActivation();
        return true;
    }

    protected void toggleActivation() {
        itemView.setActivated(adapter.isSelected(getAdapterPosition()));
    }

    public interface OnListItemClickListener {
        boolean onListItemClick(int position);
        void onListItemLongClick(int position);
    }

}
```

So you can easily override the `toggleActivation` to add whatever you want and implements the `OnListItemClickListener` on your `Activiy`/`Fragment`

**Comment by davideas** [comment]
Yes, this can be useful, indeed. I need to try it.

**Comment by danielpassos** [comment]
@davideas I have that in my locally version of this library, let me send a PR to that ;)

**Comment by davideas** [comment]
Available from 5.0.0-b1.

**Comment by davideas** [comment]
FlexibleViewHolder is heavily revisited in 5.0.0-b2 in order to support drag&drop and swipe actions
